connect/hwinfo.go

```diff
@@ -53,3 +62,29 @@ func findCloudProvider(b []byte) string {
 	}
 	return ""
 }
+
+func hypervisor() (string, error) {
+	output, err := execute([]string{"systemd-detect-virt", "-v"}, false, []int{0, 1})
+	if err != nil {
+		return "", err
+	}
+	if bytes.Equal(output, []byte("none")) {
+		return "", nil
+	}
+	return string(output), nil
+}
+
+func uuid() (string, error) {
+	if fileExists("/sys/hypervisor/uuid") {
+		content, err := os.ReadFile("/sys/hypervisor/uuid")
+		if err != nil {
+			return "", err
+		}
+		return string(content), nil
+	}
+	output, err := execute([]string{"dmidecode", "-s", "system-uuid"}, false, nil)
+	if err != nil {
+		return "", err
+	}
```

Comment:
TODO need to check for ['Not Settable', 'Not Present']
https://github.com/SUSE/connect/blob/master/lib/suse/connect/hwinfo/x86.rb#L34

Reply:
done